### PR TITLE
Enable push for update target, create/update/delete ds

### PR DESCRIPTION
**Pull request by gaurav-sahay**
The SP UI reflects updates of the repository automatically without the need of users pressing refresh buttons.
Push enabled for:
    Target update
    Added, deleted or changed distribution sets.

Signed-off-by: Asharani <asharani.murugesh@in.bosch.com>

### CODE REVIEW

**Comment by kaizimmerm** [comment]
Can you please solve the merge conflicts?

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionSetUpdateEvent.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ *
+ */
+public class DistributionSetUpdateEvent extends AbstractBaseEntityEvent<DistributionSet> {
+
+    private static final long serialVersionUID = 1L;
+
+    /**
+     * Constructor.
+     * 
+     * @param tag
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionSetUpdateEvent.java]: wrong javadoc

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/TargetUpdatedEvent.java [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+/**
+ * Defines the {@link AbstractBaseEntityEvent} of updating a {@link Target}.
+ *
+ */
+public class TargetUpdatedEvent extends AbstractBaseEntityEvent<Target> {
+
+	private static final long serialVersionUID = 5665118668865832477L;
+
+	public TargetUpdatedEvent(Target baseEntity) {
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/TargetUpdatedEvent.java]: missing javadoc

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDistributionSetManagement.java [hunk 3]
```diff
@@ -510,21 +520,23 @@ public DistributionSetMetadata updateDistributionSetMetadata(final DistributionS
...
+        final DistributionSet latestDistributionSet = findDistributionSetById(metadata.getDistributionSet().getId());
+        entityManager.merge((JpaDistributionSet) latestDistributionSet).setLastModifiedAt(0L);
         return distributionSetMetadataRepository.save(metadata);
     }
 
     @Override
     @Transactional(isolation = Isolation.READ_UNCOMMITTED)
     @Modifying
     public void deleteDistributionSetMetadata(final DistributionSet distributionSet, final String key) {
+        final DistributionSet latestDistributionSet = findDistributionSetById(distributionSet.getId());
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDistributionSetManagement.java]: a method for this two of lines would be good, there are multiple used with javadoc.

```
private void touch(final DistributionSet ds)
{
final DistributionSet latestDistributionSet = findDistributionSetById(distributionSet.getId());
entityManager.merge((JpaDistributionSet) latestDistributionSet).setLastModifiedAt(0L);
}

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/DescriptorEventDetails.java [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+	enum ActionType {
+		CREATE, UPDATE;
+	}
+
+	private DescriptorEvent descriptorEvent;
+
+	private ActionType actiontype;
+
+	public DescriptorEventDetails(ActionType actionType,
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/DescriptorEventDetails.java]: Missing javadoc

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java [hunk 5]
```diff
@@ -0,0 +1,42 @@
...
+import java.util.stream.Collectors;
+
+import org.eclipse.hawkbit.repository.eventbus.event.AbstractPropertyChangeEvent;
+import org.eclipse.hawkbit.repository.model.TenantAwareBaseEntity;
+import org.eclipse.persistence.descriptors.DescriptorEvent;
+import org.eclipse.persistence.internal.sessions.ObjectChangeSet;
+import org.eclipse.persistence.queries.UpdateObjectQuery;
+import org.eclipse.persistence.sessions.changesets.DirectToFieldChangeRecord;
+
+public class EntityPropertyChangeHelper<T extends TenantAwareBaseEntity>  {
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java]: Missing javadoc

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java [hunk 6]
```diff
@@ -0,0 +1,42 @@
...
+    
+    
+    public static <T extends TenantAwareBaseEntity> Map<String, AbstractPropertyChangeEvent<T>.Values> getChangeSet(
```
Comment (michahirsch) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java]: Missing javadoc

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/DistributionSetMetadatadetailslayout.java [hunk 7]
```diff
@@ -40,38 +40,36 @@
...
+    private Long selectedDistSetId;
+
+    /**
+     * 
+     * @param i18n
+     * @param permissionChecker
+     * @param distributionSetManagement
+     * @param dsMetadataPopupLayout
+     */
     public void init(final I18N i18n, final SpPermissionChecker permissionChecker,
```
Comment (michahirsch) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/DistributionSetMetadatadetailslayout.java]: missing javadoc and missing parameter `entityFactory`

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionDeletedEvent.java [hunk 8]
```diff
@@ -0,0 +1,36 @@
...
+ * {@link DistributionSet}.
+ */
+public class DistributionDeletedEvent extends AbstractDistributedEvent {
+    private static final long serialVersionUID = -3308850381757843098L;
+    private final Long distributionId;
+
+    /**
+     * @param tenant
+     *            the tenant for this event
+     * @param distributionSetId
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionDeletedEvent.java]: Use distributionIds in javadoc

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareManagement.java [hunk 9]
```diff
@@ -482,5 +482,23 @@ SoftwareModule findSoftwareModuleByNameAndVersion(@NotEmpty String name, @NotEmp
      */
     @PreAuthorize(SpringEvalExpressions.HAS_AUTH_UPDATE_REPOSITORY)
     SoftwareModuleType updateSoftwareModuleType(@NotNull SoftwareModuleType sm);
+    
+    /**
+     * finds all meta data by the given software module id.
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareManagement.java]: Always start sentence in upper case

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetManagement.java [hunk 10]
```diff
@@ -286,6 +286,16 @@
 
     /**
      * finds all meta data by the given distribution set id.
+     *
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetManagement.java]: Always start sentence in upper case

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java [hunk 11]
```diff
@@ -8,79 +8,47 @@
...
-        if (event.getObject().getClass().equals(Action.class)) {
-            final Action action = (Action) event.getObject();
-            if (action.getRollout() != null) {
-                final EventBus eventBus = getEventBus();
-                final AfterTransactionCommitExecutor afterCommit = getAfterTransactionCommmitExecutor();
-                afterCommit.afterCommit(() -> eventBus.post(new ActionCreatedEvent(action)));
-            }
+        final Object object = event.getObject();
+        if (isEventAwareEntity(object)) {
+            doNotifiy(() -> ((EventAwareEntity) object).fireCreateEvent(event));
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java]: Why don't you use (EventAwareEntity<Object>) object).fireUpdateEvent(.... ?

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java [hunk 12]
```diff
@@ -8,79 +8,47 @@
...
-        } else if (event.getObject().getClass().equals(JpaRollout.class)) {
-            getAfterTransactionCommmitExecutor().afterCommit(() -> getEventBus().post(
-                    new RolloutPropertyChangeEvent((Rollout) event.getObject(), getChangeSet(Rollout.class, event))));
-        } else if (event.getObject().getClass().equals(JpaRolloutGroup.class)) {
-            getAfterTransactionCommmitExecutor().afterCommit(
-                    () -> getEventBus().post(new RolloutGroupPropertyChangeEvent((RolloutGroup) event.getObject(),
-                            getChangeSet(RolloutGroup.class, event))));
+        final Object object = event.getObject();
+        if (isEventAwareEntity(object)) {
+            doNotifiy(() -> ((EventAwareEntity) object).fireUpdateEvent(event));
```
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java]: Why don't you use (EventAwareEntity<Object>) object).fireUpdateEvent(.... ?

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java (hunk 11 above)
Comment (Jonkno) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/EntityPropertyChangeListener.java]: Why don't you use EventAwareEntity<Object> ?

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionSetUpdateEvent.java (hunk 1 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionSetUpdateEvent.java]: done.

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/TargetUpdatedEvent.java (hunk 2 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/TargetUpdatedEvent.java]: done.

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDistributionSetManagement.java (hunk 3 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/JpaDistributionSetManagement.java]: done.

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/DescriptorEventDetails.java (hunk 4 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/DescriptorEventDetails.java]: done.

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java (hunk 5 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java]: done.

File: hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java (hunk 6 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-jpa/src/main/java/org/eclipse/hawkbit/repository/jpa/model/helper/EntityPropertyChangeHelper.java]: done.

File: hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/DistributionSetMetadatadetailslayout.java (hunk 7 above)
Comment (gaurav-sahay) [inline hawkbit-ui/src/main/java/org/eclipse/hawkbit/ui/common/detailslayout/DistributionSetMetadatadetailslayout.java]: done.

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionDeletedEvent.java (hunk 8 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/eventbus/event/DistributionDeletedEvent.java]: done.

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareManagement.java (hunk 9 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/SoftwareManagement.java]: done.

File: hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetManagement.java (hunk 10 above)
Comment (gaurav-sahay) [inline hawkbit-repository/hawkbit-repository-api/src/main/java/org/eclipse/hawkbit/repository/DistributionSetManagement.java]: done.

**Pull request merged**